nucypher/blockchain/eth/interfaces.py

```diff
@@ -155,6 +156,7 @@ def __init__(self,
         #
 
         self.w3 = NO_BLOCKCHAIN_CONNECTION
```

Comment:
We might be able to condense this API further and remove this attribute entirely.